samples/core/Multitenancy/exampleschema/Program.cs

```diff
@@ -0,0 +1,182 @@
+}
+
+using (var ctx = factory.CreateDbContext())
+{
+    // create database first time
+    if (ctx != null && ctx.Database.EnsureCreated())
+    {
+        var tenantAContacts = new[] { "Jane", "John", "Joe" };
+        var tenantBContacts = new[] { "Diane", "DeeDee", "Dan" };
+        ctx.TenantAContacts.AddRange(
+            tenantAContacts.Select(c => new TenantAContact { Name = c }));
+        ctx.TenantBContacts.AddRange(
+            tenantBContacts.Select(c => new TenantBContact { Name = c }));
+        ctx.SaveChanges();
+    }
+}
+
+var provider = serviceProvider.GetService<TenantProvider>();
+
+if (provider == null)
+{
+    throw new Exception("Provider should not be null.");
+}
+
+foreach (var tenant in tenants)
+{
+    Console.WriteLine($"Setting tenant to: {tenant}");
+    provider.Tenant = tenant;
+    using var ctx = factory.CreateDbContext();
+    var contacts = ctx.Contacts.OrderBy(c => c.Name).Select(c => c.Name);
+    var contactList = string.Join(", ", contacts);
+    Console.WriteLine($"Loaded contacts: {contactList}");
+}
+
+Console.WriteLine("Alternate approach.");
+
+var alternateFactory = serviceProvider
+    .GetService<IDbContextFactory<AlternateContactContext>>();
+
+if (alternateFactory == null)
+{
+    throw new Exception("Alternate factory should not be null.");
+}
+
+foreach (var tenant in tenants)
+{
+    Console.WriteLine($"Setting tenant to: {tenant}");
+    provider.Tenant = tenant;
+    using var ctx = alternateFactory.CreateDbContext();
+    var contacts = ctx.Contacts.OrderBy(c => c.Name).Select(c => c.Name);
+    var contactList = string.Join(", ", contacts);
+    Console.WriteLine($"Loaded contacts: {contactList}");
+}
+
+namespace ExampleSchema
+{
+    /// <summary>
+    /// Simple tenant service.
+    /// </summary>
+    public class TenantProvider
+    {
+        public string Tenant { get; set; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Contact.
+    /// </summary>
+    public class Contact
+    {
+        public Guid Id { get; set; } = Guid.NewGuid();
+        public string Name { get; set; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Derived class for tenant-specific support.
+    /// </summary>
+    public class TenantAContact : Contact { }
+
+    /// <summary>
+    /// Derived class for tenant-specific support.
+    /// </summary>
+    public class TenantBContact : Contact { }
+
+    /// <summary>
+    /// Using IQueryable.
+    /// </summary>
+    public class ContactContext : DbContext
+    {
+        private readonly string _tenant;
+
+        public ContactContext(
+            TenantProvider provider,
+            DbContextOptions<ContactContext> opts)
+            : base(opts) => _tenant = provider.Tenant;
+
+        public DbSet<TenantAContact> TenantAContacts { get; set; } = null!;
+        public DbSet<TenantBContact> TenantBContacts { get; set; } = null!;
+
+        public IQueryable<Contact> Contacts => _tenant == "TenantA" ?
```

Comment:
I'm not too sure about this approach... This requires modifying application code each time a tenant is added to the system, which seems problematic.

It's probably possible to do something fancy, i.e. get the full tenant list from some config file and then dynamically configure all the right tables in the context - but that's more complex than what's being shown here...

Are we sure we want to show this? The schema-per-tenant approach below is probably preferable in almost all cases anyway (unless maybe when there's no schemas like in SQLite)